test/unit/trycatch-test.js

```diff
@@ -1,50 +1,51 @@
+import ApiSettings from 'api/api-settings';
 import { tryCatch, JwError } from 'utils/trycatch';
 
 describe('trycatch', function() {
 
     it('defines', function() {
-        assert.equal(typeof tryCatch, 'function', 'trycatch function is defined');
-        assert.equal(typeof JwError, 'function', 'Error function is defined');
+        expect(typeof tryCatch, 'trycatch function is defined').to.equal('function');
+        expect(typeof JwError, 'Error function is defined').to.equal('function');
     });
 
     it('should not throw Error for valid call', function() {
-        var value = tryCatch(function() {
+        const value = tryCatch(function() {
             return 1;
         });
 
-        assert.strictEqual(value, 1, 'returns value returned by function argument when no exception is thrown');
+        expect(value, 'returns value returned by function argument when no exception is thrown').to.equal(1);
     });
 
-    it.skip('should throw custom Error on catch', function() {
-        var trycatchThrow = function() {
-            tryCatch(function() {
-                throw new Error('Danger, Danger, Will Robinson!');
-            });
-        };
+    it('should throw custom Error on catch', function() {
+        const jwError = tryCatch(function() {
+            throw new Error('Danger, Danger, Will Robinson!');
+        });
 
-        expect(trycatchThrow).to.throw(JwError);
+        expect(jwError.name).to.equal('');
+        expect(jwError.message).to.equal('Danger, Danger, Will Robinson!');
```

Comment:
This change is failing IE 11. `expect(function).to.throw(class);` is what we want. `name` is an Error property not present in IE11 and not something we need to test.

Reply:
Verifying the message prop is fine since that's something we use.

Reply:
@robwalch using a flag to test is the exception is thrown since it is caught so `to.throw` would not work. Rather going to add a check that shows that it is not being thrown.

Reply:
I see. It throws in the next test when the `debug` is enabled, but returns a `JwError` when `debug` is not enabled. The `JwError` instance was not assigned a name because `function.name` is undefined in IE 11.